File: Editorial/0092_Reverse_Linked_List_II/Go/2_Iterative_Link_Reversal.go
func reverseBetween(head *ListNode, m int, n int) *ListNode {
    if head == nil {
        return nil
    }
    var cur, prev *ListNode = head, nil
    for ; m > 1; m, n = m-1, n-1 {
        prev = cur
        cur = cur.Next
    }
    var con, tail *ListNode = prev, cur
    var third *ListNode = nil
    for ; n > 0; n-- {
        third = cur.Next
        cur.Next = prev
        prev = cur
        cur = third
    }
    if con != nil {
        con.Next = prev
    } else {
        head = prev
    }
    tail.Next = cur
    return head
}

File: Editorial/0000-0100/0092_Reverse_Linked_List_II/JavaScript/2_Iterative_Link_Reversal.js
var reverseBetween = function (head, m, n) {
    if (head === null) {
        return null;
    }
    let cur = head,
        prev = null;
    while (m > 1) {
        prev = cur;
        cur = cur.next;
        m--;
        n--;
    }
    let con = prev,
        tail = cur;
    let third = null;
    while (n > 0) {
        third = cur.next;
        cur.next = prev;
        prev = cur;
        cur = third;
        n--;
    }
    if (con !== null) {
        con.next = prev;
    } else {
        head = prev;
    }
    tail.next = cur;
    return head;
};
